modules/Bio/EnsEMBL/Compara/DBSQL/MemberAdaptor.pm

```diff
@@ -112,6 +103,51 @@ sub fetch_by_stable_id {
     }
 }
 
+=head2 fetch_by_stable_id_GenomeDB
+  Arg [1]       : string $stable_id
+  Arg [2]       : integer $genome_db_id or Bio::EnsEMBL::Compara::GenomeDB object
+  Example       : my $member = $ma->fetch_by_stable_id_GenomeDB("O93279", $genome_db);
+  Description   : Fetches the member corresponding to this $stable_id and $genome_db.
+                  This method accepts versioned stable IDs, such as ENSG00000157764.12
+  Returntype    : Bio::EnsEMBL::Compara::Member object
+  Exceptions    : throws if $stable_id and/or $genome_db is undef or has no dbID
+=cut
+
+sub fetch_by_stable_id_GenomeDB {
+    my ( $self, $stable_id, $genome_db ) = @_;
+
+    throw( "MemberAdaptor::fetch_by_stable_id_GenomeDB() must have a stable_id" ) unless $stable_id;
+    throw( "MemberAdaptor::fetch_by_stable_id_GenomeDB() must have a genome_db" ) unless $genome_db;
+    my $genome_db_id;
+    if ( $genome_db and ( $genome_db =~ /^\d+$/ ) ) {
+        $genome_db_id = $genome_db;
+    }
+    else {
+        assert_ref( $genome_db, 'Bio::EnsEMBL::Compara::GenomeDB', 'genome_db' );
+        $genome_db_id = $genome_db->dbID;
+        if ( !$genome_db_id ) {
+            throw( "[$genome_db] does not have a dbID" );
+        }
+    }
+
+    $self->bind_param_generic_fetch( $stable_id, SQL_VARCHAR );
+    $self->bind_param_generic_fetch( $genome_db_id, SQL_INTEGER );
+    my $m = $self->generic_fetch_one( 'm.stable_id = ? AND m.genome_db_id = ?' );
+    return $m if $m;
+
```

Comment:
I would add a comment explaining in which condition you are going to the last bit of code

Reply:
Do you mean for the versioned stable_ids? For example:
```suggestion
    # In the case of versioned stable_ids $m is not returned so continue with a versioned expectation
```

Reply:
> Do you mean for the versioned stable_ids? For example:

yes